whylabs_toolkit/monitor/diagnoser/converters/granularity.py

```diff
@@ -25,16 +11,17 @@ def calculate_num_batches(interval: str, granularity: str) -> int:
     except ValueError:
         end_date = start_date + parse_duration(end)
 
-    # Calculate the difference based on the granularity
+    # Calculate the (somewhat approximate) difference based on the granularity
+    # Truncates to whole batches, ignores leap seconds
     if granularity == "hourly":
-        difference = relativedelta(end_date, start_date).days * 24 + relativedelta(end_date, start_date).hours
+        difference = (end_date - start_date).total_seconds() / 3600
     elif granularity == "daily":
-        difference = relativedelta(end_date, start_date).days
+        difference = (end_date - start_date).total_seconds() / (3600 * 24)
     elif granularity == "weekly":
-        difference = relativedelta(end_date, start_date).weeks
+        difference = (end_date - start_date).total_seconds() / (3600 * 24 * 7)
     elif granularity == "monthly":
-        difference = relativedelta(end_date, start_date).months
+        difference = (end_date.year - start_date.year) * 12 + end_date.month - start_date.month
```

Comment:
I don't get why you need to multiply the year difference by 12 here

Reply:
Turns out that `.month` gives you the mod so its always between 0 and 11. So to get the full set of months you have to calculate the contribution from `.year` too